employee-rostering-server/src/test/java/org/optaweb/employeerostering/server/solver/SolverTest.java

```diff
@@ -83,46 +91,141 @@ public void testFeasibleSolution() {
         assertTrue(roster.getShiftList().stream().anyMatch(s -> s.getEmployee() != null));
     }
 
-    @Test(timeout = 600000)
-    public void testContractConstraints() {
-        HardMediumSoftLongScoreVerifier<Roster> scoreVerifier = getScoreVerifier();
-
-        AtomicLong idGenerator = new AtomicLong(1L);
+    private static class ShiftBuilder {
+
+        OffsetDateTime firstShiftStartTime;
+        Duration lengthOfShift = Duration.ofHours(9);
+        Duration durationBetweenShifts = Duration.ofDays(1);
+        Spot shiftSpot;
+        AtomicLong idGenerator;
+
+        public ShiftBuilder(AtomicLong idGenerator) {
+            this.idGenerator = idGenerator;
+        }
+
+        public ShiftBuilder startingAtDate(OffsetDateTime startDate) {
+            this.firstShiftStartTime = startDate;
+            return this;
+        }
+
+        public ShiftBuilder withShiftLength(Duration duration) {
+            this.lengthOfShift = duration;
+            return this;
+        }
+
+        public ShiftBuilder withTimeBetweenShifts(Duration duration) {
+            this.durationBetweenShifts = duration;
+            return this;
+        }
+
+        public ShiftBuilder forSpot(Spot spot) {
+            this.shiftSpot = spot;
+            return this;
+        }
+
+        public List<Shift> generateShifts(int numberOfShifts) {
+            List<Shift> out = new ArrayList<>();
+            OffsetDateTime shiftStart = firstShiftStartTime;
+
+            for (int i = 0; i < numberOfShifts; i++, shiftStart = shiftStart.plus(durationBetweenShifts)) {
+                out.add(new Shift(0, shiftSpot, shiftStart, shiftStart.plus(lengthOfShift)));
+            }
+            out.forEach(s -> s.setId(idGenerator.getAndIncrement()));
+
+            return out;
+        }
+    }
 
-        Roster roster = new Roster();
-        Tenant tenant = new Tenant("Test Tenant");
-        tenant.setId(0);
+    private LocalDate getStartDate() {
+        return LocalDate.of(2019, 5, 13);
```

Comment:
Magic numbers here and after this line.

Reply:
Actually this entire method could be replaced by a constant used by the tests.